frontend/src/components/shared/WorkflowSamplesTable/SampleTableColumns.tsx

```diff
@@ -0,0 +1,302 @@
+						/>
+					</Link>
+				)
+			)
+		},
+	},
+
+	[SampleColumnID.CONTAINER_NAME]: {
+		columnID: SampleColumnID.CONTAINER_NAME,
+		title: 'Container Name',
+		dataIndex: ['sample', 'container'],
+		render: (_, { sample }) => {
+			return (
+				sample &&
+				sample.container && (
+					<WithContainerRenderComponent
+						objectID={sample.container}
+						placeholder={<span>loading...</span>}
+						render={(container) => <span>{container.name}</span>}
+					/>
+				)
+			)
+		},
+	},
+
+	[SampleColumnID.CONTAINER_BARCODE]: {
+		columnID: SampleColumnID.CONTAINER_BARCODE,
+		title: 'Container Barcode',
+		dataIndex: ['sample', 'container'],
+		render: (_, { sample }) => {
+			return (
+				sample &&
+				sample.container && (
+					<WithContainerRenderComponent
+						objectID={sample.container}
+						placeholder={<span>loading...</span>}
+						render={(container) => <Link to={`/containers/${sample.container}`}>{container.barcode}</Link>}
+					/>
+				)
+			)
+		},
+	},
+
+	[SampleColumnID.COORDINATES]: {
+		columnID: SampleColumnID.COORDINATES,
+		title: 'Coords',
+		dataIndex: ['sample', 'coordinates'],
+	},
+
+	[SampleColumnID.VOLUME]: {
+		columnID: SampleColumnID.VOLUME,
+		title: 'Vol. (µL)',
+		dataIndex: ['sample', 'volume'],
+		align: 'right',
+		className: 'table-column-numbers',
+	},
+
+	[SampleColumnID.CONCENTRATION]: {
+		columnID: SampleColumnID.CONCENTRATION,
+		title: 'Conc. (ng/µL)',
+		dataIndex: ['sample', 'concentration'],
+		align: 'right',
+		className: 'table-column-numbers',
+		render: (conc) => (conc !== null ? parseFloat(conc).toFixed(3) : null),
+	},
+
+	[SampleColumnID.QC_FLAG]: {
+		columnID: SampleColumnID.QC_FLAG,
+		title: 'QC Flag',
+		dataIndex: ['sample', 'qc_flag'],
+		render: (_, { sample }) => {
+			if (sample) {
+				const flags = { quantity: sample.quantity_flag, quality: sample.quality_flag }
+				if (flags.quantity !== null && flags.quality !== null) {
+					return <QCFlag flags={flags} />
+				}
+			}
+			return null
+		},
+	},
+
+	[SampleColumnID.CREATION_DATE]: {
+		columnID: SampleColumnID.CREATION_DATE,
+		title: 'Creation Date',
+		dataIndex: ['sample', 'creation_date'],
+	},
+
+	[SampleColumnID.DEPLETED]: {
+		columnID: SampleColumnID.DEPLETED,
+		title: 'Depleted',
+		dataIndex: ['sample', 'depleted'],
+		render: (depleted) => <Depletion depleted={depleted} />,
+	},
+}
+
+/**
+ * Defines the set of filters that can be applied to sample columns.
+ *
+ * The object keys in the set map to the 'columnID' property of the column definition.
+ *
+ * Filter keys are defined separately, as they change depending on which endpoint
+ * is being used to retrieve samples.
+ */
+
+export const UNDEFINED_FILTER_KEY = 'UNDEFINED_FILTER_KEY'
+
+export const SAMPLE_COLUMN_FILTERS: { [key in SampleColumnID]: FilterDescription } = {
+	// Object keys map to column "columnID" properties, to match columns to filters.
+	[SampleColumnID.ID]: {
+		type: FILTER_TYPE.INPUT_OBJECT_ID,
+		key: UNDEFINED_FILTER_KEY,
```

Comment:
Why not just undefined or null ?

Reply:
I didn't want to allow undefined keys, and I decided to use that UNDEFINED_FILTER_KEY string in there to help diagnose missing key bugs (for instance, if that key shows up in redux as a filter property then it provides a good clue about what is going wrong).

Reply:
So TS made you do it ? That is a weak defense ;). I don't like the "coding for failure" strategy. It might help if your code does not work well, but your code should work well ... on its own. Coding for debugging tend to make bulkier code and it's more error prone since it does not have the fail fast quality of the undefined / null value. You can keep it if you like it, but you did not convince me ;). If undefined shows in redux you are still having an issue ;).